packages/cells/src/widget.ts

```diff
@@ -117,35 +141,52 @@ const RENDERED_CLASS = 'jp-mod-rendered';
 const DEFAULT_MARKDOWN_TEXT = 'Type Markdown and LaTeX: $ α^2 $';
 
 
+/******************************************************************************
+ * Cell
+ ******************************************************************************/
+
+
 /**
  * A base cell widget.
  */
 export
-class BaseCellWidget extends Widget {
+class Cell extends Widget {
   /**
    * Construct a new base cell widget.
    */
-  constructor(options: BaseCellWidget.IOptions) {
+  constructor(options: Cell.IOptions) {
     super();
     this.addClass(CELL_CLASS);
-    this.layout = new PanelLayout();
-
     let model = this._model = options.model;
+    let contentFactory = this.contentFactory = (
+      options.contentFactory || Cell.defaultContentFactory
+    );
+    this.layout = new PanelLayout();
 
-    let factory = this.contentFactory = options.contentFactory;
-    let editorOptions = { model, factory: factory.editorFactory };
-
-    let editor = this._editor = factory.createCellEditor(editorOptions);
-    editor.addClass(CELL_EDITOR_CLASS);
-
-    this._input = factory.createInputArea({ editor });
+    // Header
+    let header = this._header = contentFactory.createCellHeader();
+    header.addClass(CELL_HEADER_CLASS);
+    (this.layout as PanelLayout).addWidget(header);
+
+    // Input
+    let inputWrapper = this._inputWrapper = new Panel();
+    inputWrapper.addClass(CELL_INPUT_WRAPPER_CLASS);
+    let inputCollapser = this._inputCollapser = contentFactory.createCollapser();
+    inputCollapser.addClass(CELL_INPUT_COLLAPSER_CLASS);
+    let input = this._input = new InputArea({model, contentFactory });
+    input.addClass(CELL_INPUT_AREA_CLASS);
+    inputWrapper.addWidget(inputCollapser);
+    inputWrapper.addWidget(input);
     (this.layout as PanelLayout).addWidget(this._input);
+
+    // Output and footer
+    this.finishCell(options);
```

Comment:
We can't call protected methods that are meant to be subclassed from the base class constructor because the body of the subclassed constructor will not have been run yet.  (Don't call virtual methods in your constructor in C++ parlance).